src/middleware/jiraSymmetricJwtMiddleware.test.ts

```diff
@@ -0,0 +1,170 @@
+import { encodeSymmetric } from "atlassian-jwt";
+import { when } from "jest-when";
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
+import { getLogger } from "~/src/config/logger";
+import { jiraSymmetricJwtMiddleware } from "~/src/middleware/jiraSymmetricJwtMiddleware";
+import { Installation } from "~/src/models/installation";
+
+
+const logger = getLogger("jira-jwt-verify-middleware.test");
+jest.mock("models/installation");
+jest.mock("config/feature-flags");
+const testSharedSecret = "test-secret";
+
+describe("jiraSymmetricJwtMiddleware", () => {
+	let res;
+	let next;
+	let installation;
+
+	beforeEach(async () => {
+		res = {
+			locals: {
+			},
+			status: jest.fn(),
+			json: jest.fn(),
+			send: jest.fn()
+		};
+		res.status.mockReturnValue(res);
+		next = jest.fn();
+
+		installation = {
+			id: 19,
+			jiraHost,
+			clientKey: "jira-client-key",
+			enabled: true,
+			decrypt: jest.fn(() => testSharedSecret),
+			subscriptions: jest.fn().mockResolvedValue([])
+		};
+
+		jest.mocked(Installation.getForClientKey).mockImplementation((clientKey) => {
```

Comment:
Instead of mocking everything, wouldn't it be easer to create a record in Db? E.g. like we do here:

http://github.com/atlassian/github-for-jira/blob/1fc1bfb87eb30b4a805aa9990461b203d4825575/src/sqs/push.test.ts#L109-L109